dataikuapi/dss/recipe.py

```diff
@@ -1397,6 +1398,75 @@ def with_output_evaluation_store(self, mes_id):
         return self._with_output(mes_id, role="evaluationStore")
 
 
+class StandaloneEvaluationRecipeCreator(DSSRecipeCreator):
+    """
+    Builder for the creation of a new "Standalone Evaluate" recipe, from an
+    input dataset
+
+    .. code-block:: python
+
+        # Create a new standalone evaluation of a scored dataset
+
+        project = client.get_project("MYPROJECT")
+        builder = StandaloneEvaluationRecipeCreator("my_standalone_evaluation_recipe", project)
+        builder.with_input("scored_dataset_to_evaluate")
+        builder.with_output_evaluation_store(evaluation_store_id)
+
+        new_recipe = builder.create()
+
+        # Save the model parameters in the SER settings
+
+        ser_payload = dict(predictionType="BINARY_CLASSIFICATION",
+                   targetVariable="Survived",
+                   predictionVariable="prediction",
+                   isProbaAware=True,
+                   dontComputePerformance=False)
+
+        # For a classification model with probabilities, the 'probas' section can be filled with the mapping of the class and the probability column
+        # e.g. for a binary classification model with 2 columns: proba_0 and proba_1
+
+        class_0 = dict(key=0, value="proba_0")
+        class_1 = dict(key=1, value="proba_1")
+        ser_payload['probas'] = [class_0, class_1]
+
+        # Change the 'features' settings for this standalone evaluation
+        # e.g. reject the features that you do not want to use in the evaluation
+
+        feature_passengerid = dict(name="Passenger_Id", role="REJECT", type="TEXT")
+        feature_ticket = dict(name="Ticket", role="REJECT", type="TEXT")
+        feature_cabin = dict(name="Cabin", role="REJECT", type="TEXT")
+
+        ser_payload['features'] = [feature_passengerid, feature_ticket, feature_cabin]
+
+        # To set the cost matrix properly, access the 'metricParams' section of the payload and set the cost matrix weights:
+
+        ser_payload['metricParams'] = dict(costMatrixWeights=dict(tpGain=0.4, fpGain=-1.0, tnGain=0.2, fnGain=-0.5))
+
+        # Add the newly created json payload to the recipe settings and save the recipe
+        # Note that with this method, all the settings that were not explicitly set are instead set to their default value.
+
+        ser_settings = new_recipe.get_settings()
+
+        ser_settings.set_json_payload(ser_payload)
+        ser_settings.save()
```

Comment:
Sorry, one more thing I've just noticed. When you update the settings this way in the notebook, and open the recipe in the GUI, it is in a dirty state and requires to be saved.

Reply:
After digging on that item, we found the same behaviours with other recipes (evaluation, prepare). But for those, it was due to the labels, and it is another subject.
For the SER, the issue comes from completely replacing the settings dict with a new one that do not have `autoOptimizeThreshold` key. In the GUI, this key is added to the recipe desc, and therefore the recipe is considered changed.

To avoid that, and being consistent with other examples, we suggest to avoid putting the dict replacement in the documentation, and replacing it with an update of the existing dict:

`ser_settings = new_recipe.get_settings()`
`ser_json_payload = ser_settings.get_json_payload()`

`ser_json_payload['predictionType'] = "BINARY_CLASSIFICATION"`
`ser_json_payload['targetVariable'] = "Survived"`
`ser_json_payload['predictionVariable'] = "prediction"`
`ser_json_payload['isProbaAware'] = True`
`ser_json_payload['dontComputePerformance'] = False`

`ser_settings.set_json_payload(ser_json_payload)`
`ser_settings.save()`